src/client/MetaRequest.h

```diff
@@ -153,6 +153,20 @@ struct MetaRequest {
     return v == 0;
   }
 
+  void set_inode_owner_uid_gid(unsigned u, unsigned g) {
+    /* it makes sense to set owner_{u,g}id only for OPs which create inodes */
+    ceph_assert(head.op == CEPH_MDS_OP_CREATE     ||
+		head.op == CEPH_MDS_OP_OPEN       ||
+		head.op == CEPH_MDS_OP_MKNOD      ||
+		head.op == CEPH_MDS_OP_RENAMESNAP ||
+		head.op == CEPH_MDS_OP_RENAME     ||
+		head.op == CEPH_MDS_OP_MKSNAP     ||
+		head.op == CEPH_MDS_OP_MKDIR      ||
+		head.op == CEPH_MDS_OP_SYMLINK);
```

Comment:
Maybe here we can reuse the `bool Client::is_dir_operation(MetaRequest *req)`.

Reply:
hm, we can but not precisely. Set of operations in `is_dir_operation` is not the same as here. Technically we can remove this assertion at all, but I've decided to add it just as precaution.

Reply:
We can also use something like `op & CEPH_MDS_OP_WRITE`, it's close but not the same precisely too.

Reply:
Yeah, then please define a similar macro for this. Then we can reuse it in both.

Reply:
sure! Will do.

Reply:
Fixed.